datadog-setup.php

```diff
@@ -1630,6 +1645,39 @@ function ($line) {
     }
 }
 
+function map_env_to_ini($env)
+{
+    $setting = explode('_', $env, 3);
+    if (!isset($setting[0]) || $setting[0] !== 'DD' || !isset($setting[1])) {
+        return null;
+    }
+
+    $ini = 'datadog.';
+    switch ($setting[1]) {
+        case 'PROFILING':
+        case 'TRACE':
+        case 'APPSEC':
+            if (!isset($setting[2])) {
+                // this would mean $env was DD_TRACE or DD_PROFILING or DD_APPSEC
+                return null;
+            }
+            $ini .= strtolower($setting[1]) . '.' . strtolower($setting[2]);
+            break;
+        default:
+            // for cases like DD_ENV or DD_DOGSTATSD_URL
+            $ini .= strtolower($setting[1]);
+            if (isset($setting[2])) {
+                $ini .= '_' . strtolower($setting[2]);
+            }
+    }
+    foreach (get_ini_settings('', '', '') as $setting) {
```

Comment:
Why are we checking the names here? get_ini_settings() does _not_ give you all valid ini settings, but only the selected ini settings for default inclusion in the INI files. E.g. `datadog-setup.php config set DD_TRACE_REPORT_HOSTNAME=0` will fail.

Reply:
The main reason was to make sure to give guidance to users. If you do a typo it will tell you and reduce possible support cases. I get the point though. I will think about how we could tackle this as I would like to see some kind of validation in here. OTOH we could also remove that check and see if there are actually support cases.
WDYT?